### ISSUE TITLE
Allow for LiveKernels to be detected and added / removed as NotebookControllers

### ISSUE THREAD

**Issue by IanMatthewHuff** [microsoft/vscode-jupyter#5984]
![image.png](https://images.zenhubusercontent.com/5bdcd40a0faa800c749bac98/2a350ca3-3b8b-44d1-90c3-108d753b67be)

**Comment by IanMatthewHuff** [comment]
Looks like this is an actual functional issue that might take a bit of work.

When we connect to the remote machine we add our notebook kernels for installed kernels. But we also look for live running kernels. This test does the following:
Starts server
Starts a kernel with a document
Closes the document
Re-opens the document
Connects to the kernel that was previously left running
Runs a cell

But right now we only add NotebookControllers at first document open in the remote case.

There is a basic solution here, where on document open we just run our kernel finder again and addin anything missing from last time we ran it. This should pick up the new running kernels. However I did think that there might be an issue with the VS Code controller API where .dispose wouldn't remove a controller from the list. Will investigate.
